### PR TITLE
TRUNK-5652: Add NestedOrderGroups to OrderGroup

### PR DESCRIPTION
**Pull request by ODORA0**
**Issue worked on:** https://issues.openmrs.org/browse/TRUNK-5652
Added NestedOrderGroups field to the OrderGroup table

### CODE REVIEW

**Comment by ODORA0** [comment]
DB reference to the field being added to order_group table https://pastebin.com/L0Ef1Hwt

**Comment by dkayiwa** [comment]
Can you include the ticket id in your commit message as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by ODORA0** [comment]
Oh yes! I forgot it but that's done now

**Comment by dkayiwa** [comment]
Are you sure?

**Comment by ODORA0** [comment]
Yes, now I think I have! If am not mistaken the ticket id for this sub task is TRUNK-5652 and the commit message would be in this case the ticket's title that I have used, correct me if am still wrong.

**Comment by ODORA0** [comment]
Yes I see the conflicts, as far as I know, the liquibase changesets are as they have to be since I only added a nested_order_groups field of concept type

**Comment by mozzy11** [comment]
@ODORA0  , Merge conflicts are cause primarily because your origin wasnt updated with the latest changes from upstream. Can you first pull the latest changes and push again 

https://wiki.openmrs.org/display/docs/Using+Git

**Comment by dkayiwa** [comment]
@ODORA0 did you see the merge conflicts?

**Comment by ODORA0** [comment]
I have tests seriously failing and am not sure why they are, I think I made the right changes, could I be guided @dkayiwa @mozzy11

**Comment by gitcliff** [comment]
@ODORA0 i have been looking at the thread https://talk.openmrs.org/t/chemotherapy-ordering-data-design/19133/8 and dont see the NestedOderGroup metioned, last time @dkayiwa requested me to confirm from  [~rdahlman3] about this issue but she has never responded....kindly have a look at the comments on https://issues.openmrs.org/browse/TRUNK-5412

**Comment by ODORA0** [comment]
@gitcliff I see that nestedOrderGroup is not mentioned , but what did they intend to mean here
`We will use group nesting to support the idea of Premedication, Chemotherapy, and Post Medication groups within a cycle. OrderGroups will also be extended via attributes to associate cycle number, the total number of cycles in the regimen, and the length of the regimen cycles` cc @dkayiwa @mseaton @rdahlman3

File: api/src/main/java/org/openmrs/OrderGroup.java [hunk 1]
```diff
@@ -228,4 +231,26 @@ public OrderGroup getParentOrderGroup() {
...
 		this.parentOrderGroup = parentOrderGroup;
 	}
+
+	/*
+	* Gets the nestedOrderGroups 
+	*
+	*@return nestedOrderGroups
+	*@since 2.3.0
+	*/
+
```
Comment (sanjulamadurapperuma) [inline api/src/main/java/org/openmrs/OrderGroup.java]: The blank line between the comment and the method can be removed

File: api/src/main/java/org/openmrs/OrderGroup.java [hunk 2]
```diff
@@ -228,4 +231,26 @@ public OrderGroup getParentOrderGroup() {
...
+		return nestedOrderGroups;
+	}
+
+	/*
+	* Sets the nestedOrderGroups 
+	*
+	*@param nestedOrderGroups to set
+	*@since 2.3.0
+	*/
+
```
Comment (sanjulamadurapperuma) [inline api/src/main/java/org/openmrs/OrderGroup.java]: Another unnecessary space

**Pull request closed without merging**